src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -300,12 +300,10 @@ public int Compare(string? string1, string? string2, CompareOptions options)
             return retVal;
         }
 
-        internal int CompareOptionIgnoreCase(ReadOnlySpan<char> string1, ReadOnlySpan<char> string2)
-        {
-            return GlobalizationMode.Invariant ?
-                Ordinal.CompareIgnoreCaseInvariantMode(ref MemoryMarshal.GetReference(string1), string1.Length, ref MemoryMarshal.GetReference(string2), string2.Length) :
+        internal int CompareOptionIgnoreCase(ReadOnlySpan<char> string1, ReadOnlySpan<char> string2) =>
+             GlobalizationMode.Invariant ?
+                InvariantModeCasing.CompareStringIgnoreCase(ref MemoryMarshal.GetReference(string1), string1.Length, ref MemoryMarshal.GetReference(string2), string2.Length) :
                 CompareStringCore(string1, string2, CompareOptions.IgnoreCase);
```

Comment:
Can InvariantModeCasing.CompareStringIgnoreCase be used regardless of InvariantMode?

Reply:
I'm not certain on the correctness of that.

But from a "size" standpoint, I believe the `InvariantModeCasing` should be able to be trimmed for apps that aren't using Invariant. If we used InvariantModeCasing regardless of InvariantMode, it would no longer be able to be trimmed.

Reply:
It will be wrong to use invariant string comparisons here. The comparison should be performed linguistically. So, we cannot change that.